CompareRaftMembers/DiffStreamOplogFilter.js

```diff
@@ -0,0 +1,240 @@
+                    return next(null, rsId);
+                }
+                return this._fetchBucketRaftSessionId(bucketName, next);
+            },
+        ], (err, rsId) => {
+            if (err) {
+                this.emit('error', err);
+                return callback();
+            }
+            // bucket not found: ignore the entry
+            if (rsId === null) {
+                return callback();
+            }
+            const raftSessionState = this.raftSessionStates[rsId];
+            // if for some reason the returned raft session ID is not
+            // managed by us (e.g. a bucket has been recently
+            // recreated with a different raft session ID), ignore the
+            // entry
+            if (!raftSessionState) {
+                return callback();
+            }
+            raftSessionState.inputBuffer.push(diffEntry);
+            this.nBufferedEntries += 1;
+            if (this.nBufferedEntries !== this.maxBufferedEntries) {
+                // we can accept more input
+                return callback();
+            }
+            // wait until the current read buffer is flushed before
+            // accepting more input
+            this.pendingTransformCallback = callback;
+            return undefined;
+        });
+    }
+
+    _flush(callback) {
+        if (this.nBufferedEntries === 0) {
+            // end the output stream if there are no entries to forward
+            this.push(null);
+            return callback();
+        }
+        this.pendingFlushCallback = callback;
+        return undefined;
+    }
+
+    _destroy() {
+        for (const rsState of Object.values(this.raftSessionStates)) {
+            rsState.oplogStream.destroy();
+        }
+    }
+
+    /**
+     * Extract the bucket name from a diff entry coming from the stream's input
+     *
+     * @param {object} diffEntry - entry coming from DiffStream
+     * @return {string} - bucket name
+     */
+    _extractBucketName(diffEntry) {
+        const nonNullEntry = diffEntry[0] || diffEntry[1];
+        const { key: fullKey } = nonNullEntry;
+        const slashIndex = fullKey.indexOf('/');
+        const bucketName = fullKey.slice(0, slashIndex);
+        return bucketName;
+    }
+
+    _requestWrapper(reqParams, cb) {
+        const req = http.request({
+            hostname: this.bucketdHost,
+            port: this.bucketdPort,
+            agent: httpAgent,
+            ...reqParams,
+        }, res => {
+            const chunks = [];
+            res.on('data', chunk => chunks.push(chunk));
+            res.once('end', () => {
+                if (res.statusCode === 404) {
+                    return cb();
+                }
+                const body = chunks.join('');
+                if (res.statusCode !== 200) {
+                    return cb(new Error(`GET ${reqParams.path} returned status ${res.statusCode}`));
+                }
+                return cb(null, { statusCode: res.statusCode, body });
```

Comment:
statusCode should always be 200 here since we return non-200s above, right? I think just 'statusCode: 200' is easier to scan through

Reply:
indeed, will change it